File: BankKycSystemdotnet/BankKycSystem/Entities/Transaction.cs

using BankKycSystem.Enums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace BankKycSystem.Entities
{
    public class Transaction
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        public decimal? Amount { get; set; }

        public TransactionType? Type { get; set; }

        public string? Description { get; set; }

        public DateTime? Timestamp { get; set; } = DateTime.Now;

        [ForeignKey("AccountId")]
        public long? AccountId { get; set; }

        public virtual BankAccount? Account { get; set; }

       
        public void SetTimestamp()
        {
            if (Timestamp == null)
            {
                Timestamp = DateTime.Now;
            }
        }
    }
}

File: bank-kyc-system/src/main/java/com/bank/project/entity/Transaction.java
package com.bank.project.entity;

import com.bank.project.enums.TransactionType;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    private TransactionType type; 

    private String description;

    private LocalDateTime timestamp = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "account_id")
    private BankAccount account;
    
    @PrePersist
    public void prePersist() {
        if (timestamp == null) {
            timestamp = LocalDateTime.now();
        }
    }
}
